File: ulib/ml/FStar_Set.ml
type 'a set = 'a BatSet.t
let empty () =  BatSet.empty
let singleton = BatSet.singleton
let union = BatSet.union
let intersect = BatSet.intersect
let complement x = BatSet.empty
let mem = BatSet.mem

(*
 * F* should not extract Set.equal
 * We should fix it, adding the following in the meantime
 *)
type ('A, 'B, 'C) equal = unit

File: fsharp/base/FStar_Set.fs
module FStar_Set

type set<'a when 'a : comparison> = Set<'a>
let empty () = Set.empty
let singleton = Set.singleton
let union = Set.union
let intersect = Set.intersect
let complement x = Set.empty // TODO
let mem = Set.contains

(*
 * F* should not extract Set.equal
 * We should fix it, adding the following in the meantime
 *)
type equal = unit
